pkg/nebula/meta_client.go

```diff
@@ -268,31 +300,31 @@ func (m *metaClient) BalanceLeader() error {
 	return nil
 }
 
-func (m *metaClient) balance(req *meta.BalanceReq) error {
+func (m *metaClient) balance(space []byte, req *meta.AdminJobReq) error {
 	log := getLog()
-	log.Info("start balance")
-	resp, err := m.client.Balance(req)
+	log.Info("start balance job")
+	resp, err := m.client.RunAdminJob(req)
 	if err != nil {
 		log.Info("balance failed")
 		return err
 	}
-	log = log.WithValues("balanceID", resp.Id)
+	log = log.WithValues("BalanceJobID", resp.GetResult_().JobID)
 	if resp.Code != nebula.ErrorCode_SUCCEEDED {
 		if resp.Code == nebula.ErrorCode_E_LEADER_CHANGED {
 			log.Info("request leader changed", "host", resp.Leader.Host, "port", resp.Leader.Port)
 			leader := fmt.Sprintf("%v:%v", resp.Leader.Host, resp.Leader.Port)
 			if err := m.reconnect(leader); err != nil {
 				return errors.Errorf("update client failed: %v", err)
 			}
-			resp, err := m.Balance(req)
+			resp, err := m.client.RunAdminJob(req)
 			if err != nil {
 				return err
 			}
 			if resp.Code != nebula.ErrorCode_SUCCEEDED {
 				return errors.Errorf("retry balance code %d", resp.Code)
 			}
-			log.Info("balance plan running now")
-			return m.BalanceStatus(resp.Id)
+			log.Info("balance job running now")
+			return m.BalanceStatus(*resp.GetResult_().JobID, space)
```

Comment:
It seems balanceStatus is a long task, do we need use async？

Reply:
Yes, this's not a ideal implementation. I will save the last job in status field.